govuk_chat_evaluation/rag_answers/invalid_json_retry.py

```diff
@@ -0,0 +1,54 @@
+import logging
+from types import MethodType
+from typing import Any, Optional, Protocol, TypeVar
+
+from pydantic import BaseModel
+
+
+class SupportsAGenerate(Protocol):
+    async def a_generate(
+        self,
+        prompt: str,
+        schema: Optional[BaseModel] = None,
+        *args: Any,
+        **kwargs: Any,
+    ) -> Any: ...
+
+
+ModelT = TypeVar("ModelT", bound=SupportsAGenerate)
+
+
+def attach_invalid_json_retry_to_model(
+    model: ModelT,
+    max_attempts: int = 3,
+) -> ModelT:
+    """Wrap model.a_generate to retry on DeepEval's invalid JSON error."""
+    if max_attempts <= 1:
+        return model
```

Comment:
Shall we raise an error here rather than try be graceful, it feels like you shouldn't be using this code if you don't want to configure some retries. 

I'd probably change the condition to be <= 0 though but if it's 0 retries still have the code do it's stuff rather than return the model.